.github/workflows/build_changelog.yml

```diff
@@ -5,7 +5,7 @@ on:
   workflow_dispatch:
 
 permissions:
-  contents: read
+  contents: write
```

Comment:
You should not have global permissions as write permissions. Rather, define the scope directly on the job.

```yaml
jobs:
  changelog:
    permissions:
      contents: write       # create temporary branch to store changelog changes
      pull-requests: write  # create PR with changelog changes
    uses: ./.github/workflows/reusable_publish_changelog.yml
```